libp2p/crypto/secp.nim

```diff
@@ -18,235 +25,154 @@ const
     ## Size of public key in octets (bytes)
 
 type
-  SkPublicKey* = distinct s.SkPublicKey
```

Comment:
hm, probably we want to keep these distinct because of the `fromRaw` vs `fromDer` distinction as well as the defaults around compressed vs non-compressed signatures that exist in libp2p - that way, it takes the generic secp interface and "adapts" it to a libp2p-centric world, which for example is subtly different from how ethereum "uses" secp